controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java

```diff
@@ -140,15 +140,23 @@ public String getScopeName() {
      * Update configuration at configurationPath.
      *
      * @param configuration new stream configuration.
-     * @return : future of boolean
+     * @return List of active segment in the stream if stream is not sealed, otherwise returns IllegalStateException.
      */
     @Override
-    public CompletableFuture<Boolean> updateConfiguration(final StreamConfiguration configuration) {
+    public CompletableFuture<List<Integer>> updateConfiguration(final StreamConfiguration configuration) {
         // replace the configurationPath with new configurationPath
-        return verifyState(() -> updateState(State.UPDATING)
-                .thenApply(x -> setConfigurationData(configuration))
-                .thenApply(x -> true),
-                Lists.newArrayList(State.ACTIVE));
+        return verifyState(() -> getActiveSegments().thenCompose(segments -> {
+            if (segments.isEmpty()) {
+                // Stream is sealed.
+                return FutureHelpers.failedFuture(
+                        StoreException.create(StoreException.Type.OPERATION_NOT_ALLOWED, "SEALED"));
+            } else {
+                // Otherwise update stream configuration.
+                return updateState(State.UPDATING).thenCompose(x -> {
+                    return setConfigurationData(configuration).thenApply(y -> segments);
+                });
+            }
+        }), Lists.newArrayList(State.ACTIVE));
```

Comment:
Yes, need to change state to UPDATING before checking the list of active segments. Will do this in next commit. 

Let me know if you see problems in the following statements.
1. I am not inclined to keep an additional state for SEALING/SEALED because these states can be inferred from the last row of HistoryTable.
2. Moreover, the additional state variable would go away once we move createStream and updateStream out of task framework, and serially execute all updates on a stream by picking events from the request stream.